providers/desec/desecProvider.go

```diff
@@ -23,13 +24,12 @@ Info required in `creds.json`:
 func NewDeSec(m map[string]string, metadata json.RawMessage) (providers.DNSServiceProvider, error) {
 	c := &desecProvider{}
 	c.creds.token = m["auth-token"]
+	c.domainIndex = map[string]uint32{}
 	if c.creds.token == "" {
 		return nil, fmt.Errorf("missing deSEC auth-token")
 	}
-
-	// Get a domain to validate authentication
-	if err := c.fetchDomainList(); err != nil {
-		return nil, err
+	if err := c.authenticate(); err != nil {
```

Comment:
Is this going to authenticate once for each domain that calls `DnsProvider()`? 

(It might not... I would test this by setting up a dnsconfig.js with 2 or more domains, and put a Printf statement in this function to see if it prints multiple times)

Please check.

Reply:
This is in the initialization function of the provider and is only called once. I've also validated this with printf.